xarray/tests/test_distributed.py

```diff
@@ -32,3 +34,35 @@ def test_dask_distributed_integration_test(loop, engine):
                     assert isinstance(restored.var1.data, da.Array)
                     computed = restored.compute()
                     assert_allclose(original, computed)
+
+
+@pytest.mark.skipif(distributed.__version__ <= '1.19.3',
+                    reason='Need recent distributed version to clean up get')
+@gen_cluster(client=True, timeout=None)
+def test_async(c, s, a, b):
+    x = create_test_data()
+    assert not dask.is_dask_collection(x)
+    y = x.chunk({'dim2': 4}) + 10
+    assert dask.is_dask_collection(y)
+    assert dask.is_dask_collection(y.var1)
+    assert dask.is_dask_collection(y.var2)
+    # assert not dask.is_dask_collection(y.var3)  # TODO: avoid chunking unnecessarily in dataset.py::maybe_chunk
```

Comment:
We could probably argue about whether `.chunk()` should chunk variables that don't use the supplied dimension. But the default behavior is chunk everything when given an empty argument, so I think it's actually correct to do it this way (it certainly makes the return value easier to understand).

Probably a better way to do this would be to construct the dataset by hand from dask arrays, e.g.,
```python
ds = Dataset({'foo': ('x', da.arange(3, chunks=(3,)), 'bar': ('x', np.arange(3))})
assert dask.is_dask_collection(ds)
assert dask.is_dask_collection(ds.foo) 
assert not dask.is_dask_collection(ds.bar) 
```

Reply:
In a distributed context there is more cost to this behavior than with the threaded scheduler because we communicate the array around the network, rather than do things immediately/locally.